### PR TITLE
Compute: support hypervisor generation for disk

### PR DESCRIPTION
**Pull request by xccc-msft**

### CODE REVIEW

File: src/ResourceManagement/Compute/Domain/IDisk.cs [hunk 1]
```diff
@@ -32,6 +32,11 @@ public interface IDisk :
         /// </summary>
         Models.OperatingSystemTypes? OSType { get; }
 
+        /// <summary>
+        /// Gets the hypervisor generation of the Virtual Machine.
```
Comment (yaohaizh) [inline src/ResourceManagement/Compute/Domain/IDisk.cs]: Should be disk?

File: src/ResourceManagement/Compute/Domain/InterfaceImpl/DiskImpl.cs [hunk 2]
```diff
@@ -306,6 +317,16 @@ Disk.Update.IUpdate Disk.Update.IWithOSSettings.WithOSType(OperatingSystemTypes
             return this.WithOSType(osType);
         }
 
+        /// <summary>
```
Comment (yaohaizh) [inline src/ResourceManagement/Compute/Domain/InterfaceImpl/DiskImpl.cs]: Same as the above

File: src/ResourceManagement/Compute/Domain/IDisk.cs (hunk 1 above)
Comment (xccc-msft) [reply src/ResourceManagement/Compute/Domain/IDisk.cs]: The description is provided by service team within DiskInner. The hypervisor generation is related to virtual machine while it is applicable to OS disk only.

**Comment by weidongxu-microsoft** [comment]
Should have same in definition flow?

**Pull request merged**